### PR TITLE
Chapter 9: Advanced Login

### PR DESCRIPTION
**Pull request by trinhdvt**
Chapter 9: Advanced Login

- [x] Add `remember me` feature 

Ảnh chụp Màn hình 2022-03-04 lúc 12 07 50

### CODE REVIEW

**Comment by trinhdvt** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -10,10 +12,41 @@ class User < ApplicationRecord
...
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```
Comment (vanvtt-0952) [inline app/models/user.rb:35]:
```suggestion
    update_column :remember_digest, User.digest(remember_token)
```

File: app/models/user.rb [hunk 2]
```diff
@@ -10,10 +12,41 @@ class User < ApplicationRecord
...
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
+  end
+
+  def authenticate_token? remember_token
+    return false if remember_digest.nil?
```
Comment (vanvtt-0952) [inline app/models/user.rb:43]:
```suggestion
    return false unless remember_digest.present?
```

File: app/models/user.rb [hunk 3]
```diff
@@ -10,10 +12,41 @@ class User < ApplicationRecord
...
+    end
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
+
+  def forget
+    update_attribute(:remember_digest, nil)
```
Comment (vanvtt-0952) [inline app/models/user.rb:39]:
```suggestion
    update_column :remember_digest, nil
```

**Comment by trinhdvt** [comment]
ready

**Pull request merged**